x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/Rollup.java

```diff
@@ -165,8 +167,8 @@ public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settingsToUse) {
         FixedExecutorBuilder indexing = new FixedExecutorBuilder(
             settingsToUse,
             Rollup.TASK_THREAD_POOL_NAME,
-            1,
-            -1,
+            ThreadPool.searchOrGetThreadPoolSize(EsExecutors.allocatedProcessors(settingsToUse)),
```

Comment:
Is this threadpool also used by the rollup v1 jobs? I guess in the not-so-long future rollup v1 should not exist
but can this change break things for users who already run rollup v1? I mean, maybe rollup v1 concurrency model assumes that only a single task will run at any given time. In this case, increasing concurrency would cause all sorts of problems. 

I think that rollup v1 and downsampling should have their own thread pools. Then when we deprecate rollup v1, we will remove its threadpool. Can we make sure that this is the case?

Reply:
All these transport actions use different thread pools...so I honestly don't think that thread pool is used by old rollup jobs.

```
return Arrays.asList(
            new ActionHandler<>(RollupSearchAction.INSTANCE, TransportRollupSearchAction.class),
            new ActionHandler<>(PutRollupJobAction.INSTANCE, TransportPutRollupJobAction.class),
            new ActionHandler<>(StartRollupJobAction.INSTANCE, TransportStartRollupAction.class),
            new ActionHandler<>(StopRollupJobAction.INSTANCE, TransportStopRollupAction.class),
            new ActionHandler<>(DeleteRollupJobAction.INSTANCE, TransportDeleteRollupJobAction.class),
            new ActionHandler<>(GetRollupJobsAction.INSTANCE, TransportGetRollupJobAction.class),
            new ActionHandler<>(GetRollupCapsAction.INSTANCE, TransportGetRollupCapsAction.class),
            new ActionHandler<>(GetRollupIndexCapsAction.INSTANCE, TransportGetRollupIndexCapsAction.class),
            new ActionHandler<>(XPackUsageFeatureAction.ROLLUP, RollupUsageTransportAction.class),
            new ActionHandler<>(XPackInfoFeatureAction.ROLLUP, RollupInfoTransportAction.class),
            new ActionHandler<>(DownsampleIndexerAction.INSTANCE, TransportDownsampleIndexerAction.class),
            new ActionHandler<>(DownsampleAction.INSTANCE, TransportDownsampleAction.class)
        );
```

Reply:
besides the last 2, which are for downsampling (opposed to rollup).